### PR TITLE
[ci.yaml] Prefix devicelab targets with the phone used

### PR DESCRIPTION
**Pull request by CaseyHillers**
This ensures we can generate these builders correctly in the LUCI config (specifying, `linux_android`, `mac_ios32`). Most of the builders already do this, so it's making everything consistent.

https://github.com/flutter/flutter/issues/84998

https://flutter-review.googlesource.com/c/infra/+/16361 is the related LUCI config

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [X] All existing and new tests are passing.

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -1206,234 +1206,234 @@ targets:
...
+    builder: Linux_android flutter_gallery_android__compile
     presubmit: false
     properties:
       tags: >
         ["devicelab"]
     scheduler: luci
 
-  - name: Linux flutter_gallery_sksl_warmup__transition_perf_e2e
-    builder: Linux flutter_gallery_sksl_warmup__transition_perf_e2e
-    bringup: true # Flaky https://github.com/flutter/flutter/issues/83298
```
Comment (keyonghan) [inline .ci.yaml:1300]: same here

File: .ci.yaml [hunk 2]
```diff
@@ -1206,234 +1206,234 @@ targets:
...
+    builder: Linux_android flutter_gallery__start_up
     presubmit: false
     properties:
       tags: >
         ["devicelab"]
     scheduler: luci
 
-  - name: Linux flutter_gallery__transition_perf_e2e
-    builder: Linux flutter_gallery__transition_perf_e2e
-    bringup: true # Flaky https://github.com/flutter/flutter/issues/83298
```
Comment (keyonghan) [inline .ci.yaml:1259]: Is this intended?

File: .ci.yaml (hunk 1 above)
Comment (CaseyHillers) [reply .ci.yaml:1300]: Thanks, re-added! One day my LSC script will support not overriding these :)

File: .ci.yaml [hunk 3]
```diff
@@ -2057,8 +2057,17 @@ targets:
         ["devicelab"]
     scheduler: luci
 
-  - name: Mac_ios native_ui_tests_ios32
-    builder: Mac_ios native_ui_tests_ios32
+  - name: Mac_ios32 native_ui_tests_ios32
```
Comment (CaseyHillers) [inline .ci.yaml:2052]: I also added this builder as it existed in the luci config but not here

**Review by keyonghan — APPROVED** [review]
LGTM

File: .ci.yaml (hunk 3 above)
Comment (keyonghan) [reply .ci.yaml:2052]: I am not sure what you mean here. I see `Mac_ios native_ui_tests_ios32` already exists in the yaml.

Comment (CaseyHillers) [reply .ci.yaml:2060]: I fixed this. My LSC just checked for the task name in the infra config, and over-wrote them. This overwrote some of the Mac_android builders and made them Linux_android. I reran the script, and now the diff is the expected diff.

**Pull request merged**